applications/compareRestart.hpp

```diff
@@ -0,0 +1,61 @@
+/*
```

Comment:
In the C++ introduction course I took, we made header files whenever we created classes -- so I guess it is a (bad?) habit of mine. I see the point about not exposing it to the world, but I believe I want to do almost the same when comparing for example init-files, so I might want to use the class in other tests. Also, to make unit tests, won't I need to be able to include he header file to be able to test the functions in compareRestart.cpp in a unit test cppfile?

> but in that case it should go in a separate compilation unit

what does this mean, and how do I do this?